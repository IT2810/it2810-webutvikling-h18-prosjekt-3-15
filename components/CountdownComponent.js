import React from 'react';
import { View, StyleSheet} from 'react-native';
import Todo from "./Todo";
import CountdownComponent from "./CountdownComponent";

export default class App extends React.Component {
	constructor(props){
		super(props);
	}

	render() {
		return (
			<View style={styles.container}>
				{/*Countdown screen- Move to list-view when ready*/}
				<CountdownComponent
					style={styles.countdown}
					until={2700}
					timeToShow={['M', 'S']}/>
				<Todo/>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		justifyContent: 'center',
		flexDirection: 'column',
	},
	countdown:{
		marginTop: 30,
	}
});
