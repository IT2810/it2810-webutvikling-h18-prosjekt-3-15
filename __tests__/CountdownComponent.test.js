import React from 'react';
import CountdownComponent from '../components/CountdownComponent';
import renderer from 'react-test-renderer';


it('renders correctly', ()=>{
    const tree = renderer
        .create(<CountdownComponent/>)
        .toJSON();
    expect(tree).toMatchSnapshot()
});

