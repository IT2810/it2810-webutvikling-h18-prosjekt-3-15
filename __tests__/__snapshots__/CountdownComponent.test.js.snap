// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "marginTop": 7,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "backgroundColor": "#4286f4",
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              00
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Minutes
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 5,
                  "justifyContent": "center",
                  "marginHorizontal": 2,
                },
                Object {
                  "backgroundColor": "#4286f4",
                },
                Object {
                  "height": 78,
                  "width": 69,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontVariant": Array [
                      "tabular-nums",
                    ],
                    "fontWeight": "bold",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
            >
              00
            </Text>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "marginVertical": 2,
              },
              Object {
                "fontSize": 16.666666666666668,
              },
              Object {
                "color": "#000",
              },
            ]
          }
        >
          Seconds
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Play/Pause
        </Text>
      </View>
    </View>
  </View>
</View>
`;
